src/components/organisms/WithNavigationBar/WithNavigationBar.tsx

```diff
@@ -16,17 +31,71 @@ export const WithNavigationBar: React.FunctionComponent<PropsType> = ({
   redirectionUrl,
   fullscreen,
   children,
-  hasBackButton,
-}) => (
-  <>
-    {/* @debt remove backButton from Navbar */}
-    <NavBar redirectionUrl={redirectionUrl} hasBackButton={hasBackButton} />
-    <div className={`navbar-margin ${fullscreen ? "fullscreen" : ""}`}>
-      {children}
+}) => {
+  const { user } = useUser();
+  const venueId = useVenueId();
+  const venue = useSelector(currentVenueSelectorData);
+  const navigateToHomepage = useCallback(() => {
+    const venueLink =
+      redirectionUrl ?? venueId ? venueInsideUrl(venueId ?? "") : "/";
+
+    window.location.href = venueLink;
+  }, [redirectionUrl, venueId]);
+
+  const leftComponent = (
+    <div className="nav-logos">
+      <div className="nav-sparkle-logo">
+        <div className="nav-sparkle-logo_small" onClick={navigateToHomepage} />
+      </div>
+      {venue && (
+        <>
+          <NavbarSchedule />
+          <VenuePartygoers />
+        </>
+      )}
     </div>
-    <Footer />
-  </>
-);
+  );
+
+  const rightComponent = user ? (
+    <div className="navbar-links">
```

Comment:
Moving it inside makes the prop redundant because the style indicates that this is always going to be navbar-links container. If we pass another `rightComponent` for example navigation logos this would be: a) miss-leading and b) the person passing a new `rightComponent` would have to go inside the `NavBar` and change `navbar-links` to `navbar-something-else`

Reply:
I'm okey with moving it inside and renaming the rightComponent to something named specific but that kinda makes the whole prop unnecessary and we can just move the component inside.

Reply:
Not sure I follow why it’s unnecessary? The navbar-links is seemingly one of our ‘slots’ on the navbar, and that would seem like it ‘belongs’ to the navbar and its layout. Yet the content ‘put inside that slot’ isn’t inherently tied to the navbar (as I understand it)

That’s why I suggested the above.